dashboard/app/models/level_source.rb

```diff
@@ -51,4 +51,26 @@ def self.find_identical_or_create(level, data)
       end
     end
   end
+
+  # @param [Integer] user_id The ID of the user performing the encryption.
+  # @return [String] The encrypted (with the user_id) level source ID.
+  def encrypt_level_source_id(user_id)
+    Base64.urlsafe_encode64("#{id}:#{user_id}")
```

Comment:
Maybe call this `obfuscate_level_source_id`?  "Encrypt" implies we're securing the ID.